File: tests/csharp/csharp_classes/enum_basic.cs
// vybe-test: csharp/csharp_classes/enum_basic
// origin: languages/csharp/tests/csharp/test_csharp_classes.rs

using static __Harness;

Color c = Color.Green;
__P(((int)c).ToString());
__Check("1");

enum Color { Red, Green, Blue }

public static class __Harness {
    public static string __buf = "";
    public static void __P(string s) { __buf = __buf + s + "\n"; }
    public static void __Pr(string s) { __buf = __buf + s; }
    public static void __Check(string want) {
        if (__buf != want && __buf != want + "\n") {
            Console.WriteLine("FAIL: want [" + want + "] got [" + __buf + "]");
            throw new Exception("assertion failed");
        }
    }
}

File: tests/vb/vb_types/enum_basic.vb
' vybe-test: vb/vb_types/enum_basic
' origin: languages/vb/tests/vb/test_vb_types.rs

' Vybe test harness — Visual Basic.
'
' Real VB source alongside harness/go/check.go and harness/js/check.js, the way
' test262's assert.js is JavaScript.
'
' A test's verdict is its EXIT CODE. __Check prints its diagnostic BEFORE
' throwing: an uncaught exception surfaces as `RuntimeError: [object]`, which
' says nothing at all.

Module VybeCheck
    Sub __Check(got As String, want As String)
        If got <> want Then
            Console.WriteLine("FAIL: want [" & want & "] got [" & got & "]")
            Throw New Exception("assertion failed")
        End If
    End Sub
End Module

Enum Color
    Red
    Green
    Blue
End Enum

Module M
    Sub Main()
        Dim c As Color = Color.Green
        __Check(CStr(c), "1")
    End Sub
End Module
